engine/src/test/scala/cromwell/services/KeyValueServiceActorSpec.scala

```diff
@@ -0,0 +1,186 @@
+package cromwell.services
+
+import akka.pattern.ask
+import akka.util.Timeout
+import com.typesafe.config.ConfigFactory
+import cromwell.CromwellTestkitSpec.TestWorkflowManagerSystem
+import cromwell.engine.WorkflowSourceFiles
+import cromwell.engine.backend.local.LocalBackend
+import cromwell.engine.db.slick.SlickDataAccess
+import cromwell.services.KeyValueServiceActor._
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class KeyValueServiceActorSpec extends FlatSpec with Matchers with CromwellServicesTest {
+  val workflowManagerSystem = new TestWorkflowManagerSystem
+  val actorSystem = workflowManagerSystem.actorSystem
+  val dataAccess = new SlickDataAccess(ConfigFactory.parseString(
+    s"""
+       |db.url = "jdbc:hsqldb:mem:$${slick.uniqueSchema};shutdown=false;hsqldb.tx=mvcc"
+       |db.driver = "org.hsqldb.jdbcDriver"
+       |driver = "slick.driver.HsqldbDriver$$"
+       |schema.manager = slick
+     """.stripMargin
+  ))
+  val localBackend = LocalBackend(actorSystem)
+  implicit val timeout = Timeout(5.seconds)
+  implicit val ec = actorSystem.dispatcher
+
+  val cromwellConfig = ConfigFactory.parseString(
+    s"""services: {
+        |  KeyValue: {
+        |    class: "cromwell.services.KeyValueServiceActor"
+        |    config {
+        |      option1: "value1"
+        |    }
+        |  }
+        |}
+     """.stripMargin
+  )
+
+  val sources = WorkflowSourceFiles(
+    wdlSource="""task a {command{}}
+                |workflow w {
+                |  call a
+                |  call a as b
+                |  call a as c
+                |}
+              """.stripMargin,
+    inputsJson="{}",
+    workflowOptionsJson="{}"
+  )
+
+  it should "insert a key/value" in {
+    val descriptor = makeWorkflowDescriptor(sources)
+    val kvActor = makeKeyValueActor(dataAccess, cromwellConfig, descriptor)
+    val callKey = getBackendJobDescriptorKey(descriptor, "a")
+    val put = KvPut(callKey, "k", Option("v"))
+    val get = KvGet(callKey, "k")
+
+    val future = for {
+      _ <- dataAccess.createWorkflow(descriptor, Nil, descriptor.namespace.workflow.calls, localBackend)
+      r0 <- ask(kvActor, put).mapTo[KvResponse]
+      _ = r0 shouldEqual KvPutSuccess(put)
+      r1 <- ask(kvActor, get).mapTo[KvResponse]
+      _ = r1 shouldEqual KvPair(callKey, "k", Option("v"))
+    } yield ()
+
+    Await.result(future, 2.seconds)
```

Comment:
These are ok if you prefer `Await.result`, but I've leaned towards using the trait `ScalaFutures`, as its implicit `.futureValue` theoretically allows test timeouts to be "[flexibly configured](http://doc.scalatest.org/2.2.4/index.html#org.scalatest.concurrent.Futures)" on slower systems. For example, Travis isn't as zippy as our local machines.